src/context/WalletProvider/WalletViewsRouter.tsx

```diff
@@ -1,12 +1,20 @@
 import { MemoryRouter } from 'react-router-dom'
 import { useFeatureFlag } from 'hooks/useFeatureFlag/useFeatureFlag'
+import { useWallet } from 'hooks/useWallet/useWallet'
 
 import { NewWalletViewsSwitch } from './NewWalletViews/NewWalletViewsSwitch'
 import { WalletViewsSwitch } from './WalletViewsSwitch'
 
 export const WalletViewsRouter = () => {
   const isNewWalletFlowEnabled = useFeatureFlag('NewWalletFlow')
 
+  const {
+    state: { modal },
+  } = useWallet()
+
+  // Do *not* render the modal if it's not opened. Else, effects that shouldn't *will* run and produce bugs.
```

Comment:
Currently happening in all envs, v. v. bad

Reply:
And ironically a regression risk since this touches prod paths - prefer to leave it without flag constraint with the rationale that this is a blatant bug from a functional perspective, as this has been tested, but happy to add it for sanity's sake and/or if a visual/functional regression is found.